collections/ansible_collections/demo/satellite/roles/register_host/vars/RedHat8.yml

```diff
@@ -3,3 +3,6 @@ rhsm_enabled_repos:
   - rhel-8-for-x86_64-baseos-rpms
   - rhel-8-for-x86_64-appstream-rpms
   - satellite-client-6-for-rhel-8-x86_64-rpms
+instance_name: "{{ inventory_hostname | regex_replace('_','-') }}"
```

Comment:
These variables should go in defaults/main.yml so they will be defaults for RHEL 7 and 8.

I like this approach. Can you add activation_key with these as well and remove the set_fact for activation_key in the tasks?

Reply:
OK, moved